### PR TITLE
Test script

### PR DESCRIPTION
**Pull request by ctanes**
Please do not merge yet..
I would appreciate some feedback regarding general organization.
I also have a few questions regarding tests:
1) This pipeline has multiple selections / combinations on methods such as Rapsearch or blast and best_hit or humann (although humann is not yet implemented). Should I test for each combination?
2) I am only testing for the final result and not intermediate steps for the best_hit approach, mostly because I have made the intermediate steps private. Is testing for those intermediate functions necessary?

### CODE REVIEW

**Comment by kylebittinger** [comment]
To answer your questions:
1. Short answer is no, unit tests should focus on testing individual parts rather than combinations.
2. Fine to test only the non-private interface.

Part of the issue here is that we are using a unit testing library to test the whole program from start to finish.  So, here, you are not writing unit tests -- I think tests of the whole program are called acceptance or functional tests.  This is fine, and it's okay to use a unit testing library for this purpose (IMO).  However, this is why your questions are coming up.  With typical unit testing, you're looking at one method/function in isolation, so you don't have the these issues.

As we proceed, we'll want to bring more of the codebase under test.  However, I think that pragmatism should always guide you in making decisions about what/how much to test.  There is a tradeoff between your time and the level of safety you need for a particular part of the code.  I'll work with you to establish the right balance and to design functions that are easy to test.

File: test/test_main.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+            observed = json.load(f)
+        self.assertEqual(observed.get('data', {}), EXPECTED_SUMMARY)
+
+    def test_make_index_bestHit(self):
+        # create the mock kegg file
+        kegg = tempfile.NamedTemporaryFile()
+        kegg.write(MOCK_KEGG)
+        kegg.seek(0)
+        
+        # create temp directory for the index
```
Comment (kylebittinger) [inline test/test_main.py]: Don't need this comment; it's misleading anyway

File: test/test_main.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        kegg2ko = tempfile.NamedTemporaryFile()
+        self.config['kegg_fp'] = kegg.name
+        self.config['kegg_to_ko_fp'] = kegg2ko.name
+
+        # make index
+        assignerApp = Assignment(self.config)
```
Comment (kylebittinger) [inline test/test_main.py]: Assigner or AssignmentMethod would be a better name for the class

File: test/test_main.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        r1.write(MOCK_NO_MATCH_R1)
+        r1.seek(0)
+        r2 = tempfile.NamedTemporaryFile(suffix=".fastq")
+        r2.write(MOCK_NO_MATCH_R2)
+        r2.seek(0)
+                                
+        self.run_pipeline(r1, r2, config_file)
+        
+        #check if the output is correct
+        myoutput_fp = os.path.join(self.output_dir, os.path.basename(os.path.splitext(r1.name)[0]+'.ko'))
```
Comment (kylebittinger) [inline test/test_main.py]: This is repeated several times.  I wonder if we can get this logic into the code rather than keeping it in the test.

**Comment by kylebittinger** [comment]
Overall I think this is great and OK to go in if the tests pass.  I'll wait to merge until you give me the go-ahead.

**Comment by ctanes** [comment]
I have addressed your comments and tidied up the code. I think we can merge this to the main branch.

File: test/test_main.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        # make index
+        assignerApp = Assigner(self.config)
+        assignerApp.make_index()
+
+        # check equal
+        observed = open(self.config["kegg_to_ko_fp"]).read()
+        self.assertEqual(observed.strip(), KEGG_TO_KO.strip())
+
+
+# get known sequences from KEGG to test the cases:
```
Comment (kylebittinger) [inline test/test_main.py:89]: This is great.

**Pull request merged**